### PR TITLE
adds an area description blurb system

### PR DESCRIPTION
**Pull request by Mazianni**
This PR adds a system for displaying a small blurb of text to a player when they enter an area. This only happens **once** per area.

### CODE REVIEW

File: code/game/area/areas.dm [hunk 1]
```diff
@@ -307,12 +309,24 @@ var/global/list/mob/living/forced_ambiance_list = new
...
 
+/area/proc/do_area_blurb(var/mob/living/L)
+	if(isnull(description))
+		return
+
+	if(!(L && L.client && L.get_preference_value(/datum/client_preference/area_info_blurb) == PREF_YES))
+		return 
+	
+	if(!(L.client in blurbed_stated_to))
+		blurbed_stated_to += L.client
```
Comment (Lohikar) [inline code/game/area/areas.dm]: Yeah, don't shove client refs into long lived lists please. Clients that log out and back in will be shown the message again.

Add the ckey instead.

Comment (Mazianni) [reply code/game/area/areas.dm]: Addressed.

File: code/game/area/areas.dm [hunk 2]
```diff
@@ -307,12 +309,24 @@ var/global/list/mob/living/forced_ambiance_list = new
...
 	return ..()
 
+/area/proc/do_area_blurb(var/mob/living/L)
+	if(isnull(description))
+		return
+
+	if(!(L && L.client && L.get_preference_value(/datum/client_preference/area_info_blurb) == PREF_YES))
+		return 
+	
+	if(!(lowertext(L.ckey) in blurbed_stated_to))
```
Comment (MistakeNot4892) [inline code/game/area/areas.dm]: ckeys are always lowercase, they're always the same as output from `ckey()`.

File: code/game/area/areas.dm [hunk 3]
```diff
@@ -307,12 +309,24 @@ var/global/list/mob/living/forced_ambiance_list = new
...
 /area/Exited(A)
 	if(isliving(A))
 		clear_ambience(A)
 	return ..()
 
+/area/proc/do_area_blurb(var/mob/living/L)
+	if(isnull(description))
+		return
+
+	if(!(L && L.client && L.get_preference_value(/datum/client_preference/area_info_blurb) == PREF_YES))
```
Comment (MistakeNot4892) [inline code/game/area/areas.dm]: `if(L?.get_preference_value(foo) != PREF_YES)` would be considerably shorter.

File: code/modules/client/preference_setup/global/preferences.dm [hunk 4]
```diff
@@ -333,3 +333,12 @@ var/global/list/_client_preferences_by_type
...
 	flags = R_ADMIN|R_DEBUG
+
+/********************
+* Area Info Blurb *
+********************/
+
+/datum/client_preference/area_info_blurb
+	description ="Show area information"
+	key = "AREA_INFO"
+	default_value = PREF_YES
```
Comment (MistakeNot4892) [inline code/modules/client/preference_setup/global/preferences.dm:344]: Honestly I would prefer this to default to PREF_NO, though I'm not strongly opinionated either way.

Comment (Mazianni) [reply code/modules/client/preference_setup/global/preferences.dm:344]: I have a feeling this will never be seen if it defaults to No, tbh.

**Review by MistakeNot4892 — APPROVED** [review]

**Pull request merged**